rust/core-lib/src/file.rs

```diff
@@ -244,6 +269,16 @@ where
     File::open(path).and_then(|f| f.metadata()).and_then(|meta| meta.modified()).ok()
 }
 
+#[cfg(target_family = "unix")]
+/// Returns the file permissions for the file at a given path on UNIXy systems,
+/// if present.
+fn get_permissions<P>(path: P) -> Option<u32>
+where
+    P: AsRef<Path>,
```

Comment:
I prefer to put generic constraints inline, when doing so doesn't produce overly long lines, e.g.

```rust
fn get_permissions<P: AsRef<Path>>(path: P) -> Option<u32> {
// ...
```
seems cleaner, here?

Reply:
Sure, if that's what you prefer. I just always use where to be consistent.

Reply:
yep I think that's fair, but the main perk I get out of doing all these reviews is being able to impose my personal stylistic preferences on people 😉